DigitalLearningSolutions.Data/Services/LearningHubResourceService.cs

```diff
@@ -45,16 +42,15 @@ int resourceReferenceId
         {
             try
             {
-                // TODO HEEDLS-747: tackle different response codes rather than catching all exceptions
                 var upToDateResourceDetails = await learningHubApiClient.GetResourceByReferenceId(resourceReferenceId);
                 return (upToDateResourceDetails, false);
             }
-            catch (HttpRequestException e)
+            catch (LearningHubResponseException e)
             {
-                logger.LogWarning(
-                    $"Call to Learning Hub Open API failed when trying to call /Resource/{resourceReferenceId} endpoint. " +
-                    $"Using fall-back data instead. Exception: {e.Message}"

```

Comment:
I think it could still be useful for debugging if we mentioned that we're using fallback data here (and in bulk get)

Reply:
I've added the fallback data warning to the logging being done in the API client. I think that is suitable enough, but it does cover the Search endpoint. If you think that we should restrict it to a per-method basis in the LearningHubResourceService, then I can easily change it around.

Reply:
I think the problem with doing it there is that it makes the `GetStringAsync` method context-specific. It shouldn't really know what's going to happen if a request fails - that's the responsibility of LearningHubResourceService.